client/my-sites/upgrades/domain-management/name-servers/dns-templates.jsx

```diff
@@ -2,23 +2,52 @@
  * External dependencies
  */
 import React, { Component } from 'react';
-import { find } from 'lodash';
+import { find, replace } from 'lodash';
 import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
+import ButtonGroup from 'components/button-group';
 import Card from 'components/card';
+import { dnsTemplates } from 'lib/domains/constants';
 import DnsTemplateButton from './dns-template-button';
-import Office365 from '../dns/office-365';
+import EmailProvider from '../dns/email-provider';
 
 class DnsTemplates extends Component {
 	constructor( props ) {
 		super( props );
+		const { translate } = this.props;
 		this.state = {
 			currentComponentName: null,
+			currentProviderCardName: null,
 			templates: [
-				{ name: 'Office 365', Component: Office365 }
+				{
+					name: 'G Suite',
+					label: translate( 'G Suite Verification Token - from the TXT record verification' ),
+					placeholder: 'google-site-verification=...',
+					validationPattern: /^google-site-verification=\w{43}$/,
+					dnsTemplate: dnsTemplates.G_SUITE
+				},
+				{
+					name: 'Office 365',
+					label: translate( 'Office 365 Verification Token - from the TXT record verification' ),
+					placeholder: 'MS=ms...',
+					validationPattern: /^MS=ms\d{4,20}$/,
+					dnsTemplate: dnsTemplates.MICROSOFT_OFFICE365,
+					modifyVariables: ( variables ) => Object.assign(
+						{},
+						variables,
+						{ mxdata: replace( variables.domain, '.', '-' ) + '.mail.protection.outlook.com.' }
+					)
+				},
+				{
+					name: 'Zoho Mail',
+					label: translate( 'Zoho Mail CNAME zb code' ),
+					placeholder: 'zb...',
+					validationPattern: /^zb\w+$/,
```

Comment:
Even if we don't know the maximum length of the "zbcode", we should still limit it to something reasonable - otherwise, we might run into troubles when someone, for example, inputs a >256 characters long string (for whatever reason) and we try to create a CNAME based on that ;/